File: daily_action_cycle_mobile_ui/lib/core/entities/entity.dart
import 'dart:core';

class Entity {
  String? id;
  String? name;
  String? description;
  DateTime? createdAt;
  DateTime? updatedAt;
  DateTime? deletedAt;
  bool completed = false;

  Entity(this.id, this.name, this.description, this.createdAt, this.completed);
}

File: DailyActionCycle.WebAPI/DailyActionCycle.WebAPI/Entities/Abstracts/Entity.cs
namespace DailyActionCycle.Core.Entities.Abstracts;
public record Entity
{
    public Guid Id { get; init; }

    public required string Name { get; init; }

    public required string Description { get; init; }

    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    public DateTime? DeletedAt { get; set; }

    public bool Completed { get; private set; } = false;

    public void Complete() => Completed = true;
}
